autopilot/agent_test.go

```diff
@@ -1546,3 +1546,157 @@ func TestAgentSkipPendingConns(t *testing.T) {
 		t.Fatalf("agent should have attempted connection")
 	}
 }
+
+// TestAgentQuitWhenPendingConns tests that we are able to stop the autopilot
+// agent even though there are pending connections to nodes.
+func TestAgentQuitWhenPendingConns(t *testing.T) {
```

Comment:
Not a blocker, but looks like we'd really benefit from abstracting some common patterns within the tests into helper functions for easier readability.